parse/IntComplexValueRefParser.cpp

```diff
@@ -102,6 +102,30 @@ namespace parse {
                     ) [ _val = new_<ValueRef::ComplexVariable<int> >(_a, _b, _c, _f, _d, _e) ]
                 ;
 
+            ship_parts_owned
+                =   (
+                            tok.ShipPartsOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Name_token)   >    string_value_ref [ _d = _1 ] )
+                    ) [ _val = new_<ValueRef::ComplexVariable<int>>(_a, _b, _c, _f, _d, _e) ]
+                ;
+
+            ship_part_class_owned
+                = (
+                            tok.ShipPartClassOwned_ [ _a = construct<std::string>(_1) ]
+                        >-( parse::label(Empire_token) >    int_value_ref [ _b = _1 ] )
+                        >-( parse::label(Class_token)  >
```

Comment:
hmm, looks like I had the same idea before and gave up on getting it to work, and did what you've done here (for part_class_in_ship_design)

Maybe a solution here, but probably not worth the effort: http://www.doing-stuff.org/2015/04/obtaining-both-the-attribute-and-the-raw-string-in-boostspirit/